### PR TITLE
Feature/update ssss activity

### LINKED ISSUE #1199: XSigning | Trust account xSigning keys by entering Recovery Key (select file or copy)
Current UX only let you enter a passphrase
-> Need to check the key content to see if it was a passphrase or not

### PR DESCRIPTION
**Pull request by BillCarsonFr**
Fixes #1199
Support recovery key mode for SSSS access
image

image

image

### CODE REVIEW

File: vector/src/main/java/im/vector/riotx/features/crypto/quads/SharedSecuredStoragePassphraseFragment.kt [hunk 1]
```diff
@@ -41,15 +46,17 @@ class SharedSecuredStoragePassphraseFragment : VectorBaseFragment() {
...
+        val pass = getString(R.string.recovery_passphrase)
+        val key = getString(R.string.recovery_key)
+        ssss_restore_with_passphrase_warning_text.text = getString(
+                R.string.enter_secret_storage_passphrase_or_key,
+                pass,
+                key
+        )
+                .toSpannable()
+                .colorizeMatchingText(pass, colorProvider.getColorFromAttribute(android.R.attr.textColorLink))
+                .colorizeMatchingText(key, colorProvider.getColorFromAttribute(android.R.attr.textColorLink))
```
Comment (bmarty) [inline vector/src/main/java/im/vector/riotx/features/crypto/quads/SharedSecuredStoragePassphraseFragment.kt:59]: links are still not clickable, to it confuses users. See #1257 . Maybe comment this out for the moment?

File: vector/src/main/res/values/strings_riotX.xml [hunk 2]
```diff
@@ -7,6 +7,11 @@
 
     <!-- BEGIN Strings added by Valere -->
 
+    <!-- first will be replaced by recovery_passphrase, second will be replaced by recovery_key-->
+    <string name="enter_secret_storage_passphrase_or_key">Use your %1$s or Use your %2$s to continue.</string>
```
Comment (bmarty) [inline vector/src/main/res/values/strings_riotX.xml:11]:
```suggestion
    <string name="enter_secret_storage_passphrase_or_key">Use your %1$s or use your %2$s to continue.</string>
```

File: vector/src/main/java/im/vector/riotx/features/crypto/quads/SharedSecuredStoragePassphraseFragment.kt (hunk 1 above)
Comment (BillCarsonFr) [reply vector/src/main/java/im/vector/riotx/features/crypto/quads/SharedSecuredStoragePassphraseFragment.kt:59]: There will be a link at some point

File: vector/src/main/java/im/vector/riotx/features/crypto/quads/SharedSecureStorageViewModel.kt [hunk 3]
```diff
@@ -66,21 +76,129 @@ class SharedSecureStorageViewModel @AssistedInject constructor(
...
+        _viewEvents.post(SharedSecureStorageViewEvent.ShowModalLoading)
+        val decryptedSecretMap = HashMap<String, String>()
+        setState { copy(checkingSSSSAction = Loading()) }
+        viewModelScope.launch(Dispatchers.IO) {
+            runCatching {
+                val recoveryKey = action.recoveryKey
+                val keyInfoResult = session.sharedSecretStorageService.getDefaultKey()
+                if (!keyInfoResult.isSuccess()) {
+                    _viewEvents.post(SharedSecureStorageViewEvent.HideModalLoading)
+                    _viewEvents.post(SharedSecureStorageViewEvent.Error("Cannot find ssss key"))
```
[the remark this answers is not in the document]
Comment (BillCarsonFr) [reply vector/src/main/java/im/vector/riotx/features/crypto/quads/SharedSecureStorageViewModel.kt:147]: Added

**Pull request merged**